File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/clinical-new/shared/clinical-info-preview/dto/clinical-data.dto.ts
import { ClinicalCommonData_DTO } from "./clinical-common-data.dto";


export class ClinicalData_DTO extends ClinicalCommonData_DTO {
    ClinicalInformationId: number;
    ClinicalHeadingId: number;
    FieldId: number;
    InputType: string;
    FieldValue: string;
    Remarks: string;
    ClinicalHeadingName: string;
    ParentId: number;
    HeadingDisplayName: string;
    FieldCode: string;
    FieldName: string;
    FieldDisplayName: string;
    IsEditable: boolean = false;
    ClinicalOptionsData: ClinicalOptions_DTO[];
    ClinicalAnswerData: ClinicalQuestionAnswer_DTO[];
    ClinicalAnswerOptionData: ClinicalQuestionOption_DTO[];
    // IsInEditMode?: boolean = false;
    IsPrintable?: boolean = false;


}

export class ClinicalOptions_DTO extends ClinicalCommonData_DTO {
    ClinicalOptionRecordId: number;
    ClinicalInformationId: number;
    ParentHeadingId: number;
    FieldId: number;
    OptionId: number;
    Options: string;
}

export interface ClinicalQuestionAnswer_DTO extends ClinicalCommonData_DTO {
    ClinicalQuestionAnswerId: number;
    ClinicalInformationId: number;
    ClinicalHeadingId: number;
    ParentHeadingId: number;
    FieldId: number;
    QuestionId: number;
    AnswerValue: string;
    Remarks: string;
    AnswerType: string;
    Question: string;
    // IsInEditMode?: boolean;
    IsPrintable?: boolean;
    IsEditable?: boolean;
}

export interface ClinicalQuestionOption_DTO extends ClinicalCommonData_DTO {
    ClinicalAnswerOptionId: number;
    ClinicalInformationId: number;
    ClinicalHeadingId: number;
    ParentHeadingId: number;
    FieldId: number;
    QuestionId: number;
    QuestionOptionId: number;
    Remarks: string;
    QuestionOption: string;
}

File: Code/Websites/DanpheEMR/Services/Clinical_New/DTOs/ClinicalData_DTO.cs
using System;
using System.Collections.Generic;

namespace DanpheEMR.Services.Clinical_New.DTOs
{
 
    public class clinicalCommonData
    {
        public int PatientId { get; set; }
        public int PatientVisitId { get; set; }
        public DateTime CreatedOn { get; set; }
        public int CreatedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public int? ModifiedBy { get; set; }
		public string EnteredBy { get; set; }
	}
    public class ClinicalData_DTO : clinicalCommonData
    {
        public int ClinicalInformationId { get; set; }
        public int ClinicalHeadingId { get; set; }
        public int FieldId { get; set; }
        public string InputType { get; set; }
        public string FieldValue { get; set; }
        public string Remarks { get; set; }
        public string ClinicalHeadingName { get; set; }
        public int ParentId { get; set; }
        public string HeadingDisplayName { get; set; }
        public string FieldCode { get; set; }
        public string FieldName { get; set; }
        public string FieldDisplayName { get; set; }
        public bool IsEditable { get; set; }
        public List<ClinicalOptions_DTO> ClinicalOptionsData { get; set; }
        public List<ClinicalQuestionAnswer_DTO> ClinicalAnswerData { get; set; }
        public List<ClinicalQuestionOption_DTO> ClinicalAnswerOptionData { get; set; }
    }
   
    public class ClinicalOptions_DTO : clinicalCommonData
    {
        public int ClinicalOptionRecordId { get; set; }
        public int ClinicalInformationId { get; set; }
        public int? ParentHeadingId { get; set; }
        public int FieldId { get; set; }
        public int OptionId { get; set; }
        public string Options { get; set; }
    }

    public class ClinicalQuestionAnswer_DTO : clinicalCommonData
    {
        public int ClinicalQuestionAnswerId { get; set; }
        public int ClinicalInformationId { get; set; }
        public int ClinicalHeadingId { get; set; }
        public int ParentHeadingId { get; set; }
        public int FieldId { get; set; }
        public int QuestionId { get; set; }
        public string AnswerValue { get; set; }
        public string Remarks { get; set; }
        public string AnswerType { get; set; }
        public string Question { get; set; }
    }

    public class ClinicalQuestionOption_DTO : clinicalCommonData
    {
        public int ClinicalAnswerOptionId { get; set; }
        public int ClinicalInformationId { get; set; }
        public int ClinicalHeadingId { get; set; }
        public int? ParentHeadingId { get; set; }
        public int FieldId { get; set; }
        public int QuestionId { get; set; }
        public int QuestionOptionId { get; set; }
        public string Remarks { get; set; }
        public string QuestionOption { get; set; }


    }
}
